OpenSim/Sandbox/ExampleHopperDevice/buildHopperModel.cpp

```diff
@@ -0,0 +1,205 @@
+    hopper.addContactGeometry(foot);
+    hopper.addForce(contactForce);
+
+    // Create the vastus muscle and set its origin and insertion points.
+    double mclFmax = 4000., mclOptFibLen = 0.55, mclTendonSlackLen = 0.25,
+           mclPennAng = 0.;
+    auto vastus = new Thelen2003Muscle("vastus", mclFmax, mclOptFibLen,
+                                       mclTendonSlackLen, mclPennAng);
+    vastus->addNewPathPoint("origin", *thigh, Vec3(linkRadius, 0.1, 0));
+    vastus->addNewPathPoint("insertion", *shank, Vec3(linkRadius, 0.15, 0));
+    hopper.addForce(vastus);
+
+    // Attach a cylinder (patella) to the distal end of the thigh over which the
+    // vastus muscle can wrap. In the future, it will be possible to wrap over
+    // WrapObjects that are attached to PhysicalOffsetFrames; for now, we must
+    // attach the patella to the thigh Body and use the "translation" property
+    // to position the patella at the knee.
+    auto patella = new WrapCylinder();
+    patella->setName("patella");
+    patella->set_radius(0.08);
+    patella->set_length(linkRadius*2.);
+    patella->set_quadrant("x");
+    patella->set_translation(linkDistalPoint);
+    thigh->addWrapObject(patella);
+
+    // Configure the vastus muscle to wrap over the patella. We are currently
+    // using a helper method because of a bug in GeometryPath; in the future, we
+    // will be able to do: vastus->updGeometryPath().addPathWrap(*patella);
+    addPathWrapHelper(hopper, "vastus", "patella", "thigh");
+
+    // Create a controller to excite the vastus muscle.
+    auto brain = new PrescribedController();
+    brain->setActuators(hopper.updActuators());
+    double t[3] = {0.0, 2.0, 3.9}, x[3] = {0.3, 1.0, 0.1};
+    auto controlFunction = new PiecewiseConstantFunction(3, t, x);
+    brain->prescribeControlForActuator("vastus", controlFunction);
+    hopper.addController(brain);
+
+    // Create frames on the thigh and shank segments for attaching the device.
+    auto thighAttachment = new PhysicalOffsetFrame("deviceAttachmentPoint",
+                           *thigh, SimTK::Transform(Vec3(linkRadius, 0.1, 0)));
+    auto shankAttachment = new PhysicalOffsetFrame("deviceAttachmentPoint",
+                           *shank, SimTK::Transform(Vec3(linkRadius, 0, 0)));
+    hopper.addFrame(thighAttachment);
+    hopper.addFrame(shankAttachment);
+
+    // Attach geometry to the bodies and enable the visualizer.
+    auto pelvisGeometry = Brick(Vec3(pelvisSideLength/2.));
+    pelvisGeometry.setColor(Vec3(0.8, 0.1, 0.1));
+    pelvis->attachGeometry(pelvisGeometry);
+
+    auto linkGeometry = Cylinder(linkRadius, linkLength/2.);
+    linkGeometry.setColor(Vec3(0.8, 0.1, 0.1));
+    thigh->attachGeometry(linkGeometry);
+    shank->attachGeometry(linkGeometry);
+
+    hopper.setUseVisualizer(true);
+
+    // We're done! Print a list of all components in the model (for debugging).
+    if (printDebugInfo) { hopper.dumpSubcomponents(); }
+
+    return hopper;
```

Comment:
This comment is not related to the PR: 

Now that I see this, I think this would be a great example for the front of the API Guide. Similar to the Tug-of-War C++ example, it showcases many different components. The main issue I see is that the API Guide's example should showcase connectors, inputs, and outputs as well. To deal with the length of the example, it could be broken up into 3 chunks (code blocks). Perhaps "Bodies and joints,", "Actuation," "Contact forces, control, and geometry."